src/Logger.cpp

```diff
@@ -86,7 +91,16 @@ void cLogger::AttachListener(cListener * a_Listener)
 void cLogger::DetachListener(cListener * a_Listener)
 {
 	cCSLock Lock(m_CriticalSection);
-	m_LogListeners.erase(std::remove(m_LogListeners.begin(), m_LogListeners.end(), a_Listener));
+	m_LogListeners.erase(
+		std::remove_if(
+			m_LogListeners.begin(),
+			m_LogListeners.end(),
+			[=](std::unique_ptr<cListener> & a_OtherListener) -> bool
+			{
+				return a_OtherListener.get() == a_Listener;
+			}
+		)
```

Comment:
The `-> bool` is not necessary, but it adds a little clarity, and with the lambda style we use, its not overly verbose.

As for by value captures, the only thing this lambda captures is a pointer, which is usually more efficient to capture by value.